arviz/plots/backends/matplotlib/traceplot.py

```diff
@@ -117,73 +110,17 @@ def _plot_trace_mpl(
         >>> az.plot_trace(data, var_names=('theta_t', 'theta'), coords=coords, lines=lines)
 
     """
-    if divergences:
-        try:
-            divergence_data = convert_to_dataset(data, group="sample_stats").diverging
-        except (ValueError, AttributeError):  # No sample_stats, or no `.diverging`
-            divergences = False
-
-    if coords is None:
-        coords = {}
-
-    data = get_coords(convert_to_dataset(data, group="posterior"), coords)
-    var_names = _var_names(var_names, data)
+    # Set plot default backend kwargs
+    if backend_kwargs is None:
+        backend_kwargs = {}
 
-    if divergences:
-        divergence_data = get_coords(
-            divergence_data, {k: v for k, v in coords.items() if k in ("chain", "draw")}
-        )
-
-    if lines is None:
-        lines = ()
-
-    num_colors = len(data.chain) + 1 if combined else len(data.chain)
-    colors = [
-        prop
-        for _, prop in zip(
-            range(num_colors), cycle(plt.rcParams["axes.prop_cycle"].by_key()["color"])
-        )
-    ]
-
-    if compact:
-        skip_dims = set(data.dims) - {"chain", "draw"}
-    else:
-        skip_dims = set()
-
-    plotters = list(xarray_var_iter(data, var_names=var_names, combined=True, skip_dims=skip_dims))
-    max_plots = rcParams["plot.max_subplots"]
-    max_plots = len(plotters) if max_plots is None else max_plots
-    if len(plotters) > max_plots:
-        warnings.warn(
-            "rcParams['plot.max_subplots'] ({max_plots}) is smaller than the number "
-            "of variables to plot ({len_plotters}), generating only {max_plots} "
-            "plots".format(max_plots=max_plots, len_plotters=len(plotters)),
-            SyntaxWarning,
-        )
-        plotters = plotters[:max_plots]
-
-    if figsize is None:
-        figsize = (12, len(plotters) * 2)
-
-    if trace_kwargs is None:
-        trace_kwargs = {}
-
-    trace_kwargs.setdefault("alpha", 0.35)
-
-    if hist_kwargs is None:
-        hist_kwargs = {}
-    if plot_kwargs is None:
-        plot_kwargs = {}
-    if fill_kwargs is None:
-        fill_kwargs = {}
-    if rug_kwargs is None:
-        rug_kwargs = {}
-
-    hist_kwargs.setdefault("alpha", 0.35)
+    backend_kwargs = {**BACKEND_KWARG_DEFAULTS, **backend_kwargs}
 
     figsize, _, titlesize, xt_labelsize, linewidth, _ = _scale_fig_size(
-        figsize, textsize, rows=len(plotters), cols=2
+        figsize, backend_kwargs["textsize"], rows=len(plotters), cols=2
     )
+
+    # TODO: This is breaking plotting for some reason
```

Comment:
How?

Reply:
It was the linewidth thing. Comment is unneeded now and has been deleted.